src/v/security/gssapi_authenticator.cc

```diff
@@ -399,7 +400,8 @@ result<void> gssapi_authenticator::check() {
     //     return errc::invalid_credentials;
     // }
 
-    _principal = acl_principal{principal_type::user, ss::sstring{source_name}};
+    _principal = get_principal_from_name(source_name);
+
```

Comment:
@BenPope is it true that principal here is set from the alien thread, and read from the reactor thread as part of the authenticator interface? if so, we should probably (1) verify it is currently safe from races, and/or even better (2) eliminate the race by doing a sort of message passing to/from the alien thread.

Reply:
https://github.com/redpanda-data/redpanda/issues/8366